website/gatsby-node.js

```diff
@@ -15,6 +15,21 @@ exports.onCreateWebpackConfig = ({ actions }) => {
   actions.setWebpackConfig({
     resolve: {
       alias
+    },
+    /**
+     * Inspiration from https://github.com/gatsbyjs/gatsby/issues/10965. There seems to be an issue where a couple dependecies are duplicated
+     * across each page component and not properly split out into commons chunks (prettier, babel parser are the main ones)
+     */
+    optimization: {
+      splitChunks: {
+        cacheGroups: {
+          commons: {
+            name: `commons`,
+            chunks: `all`,
+            minChunks: 2
```

Comment:
Is it worth including [`reuseExistingChunk: true`](https://webpack.js.org/plugins/split-chunks-plugin/#splitchunkscachegroupscachegroupreuseexistingchunk) like in [this example](https://github.com/gatsbyjs/gatsby/issues/10965#issuecomment-453456324)?

What impact does it have on build time and bundle size?

Reply:
Doesn't seem to make any difference adding `reuseExistingChunk: true ` , both build time and bundle size is the same. The key line is `minChunks: 2`. In the Gatsby webpack config it is set as `minChunks: componentsCount > 2 ? componentsCount * 0.5 : 2,`. Which adds the chunk to commons if it's used by more than half the components (I think). In theory I feel like that should be working as prettier is being imported on every component.

Overall adding this change reduced my build time of `Building production JavaScript and CSS bundles` from ~240s to ~40s and overall bundle size from 34mb to ~2.8mb

Reply:
Cool, that's awesome!